lab_3/main.py

```diff
@@ -222,7 +359,22 @@ def save(self, name: str) -> int:
         :param name: name of the json file with .json format
         :return: 0 if profile saves, 1 if any errors occurred
         """
-        pass
+        if not isinstance(name, str):
+            return 1
+        language_profile = {}
+        freq_dict = {}
+        for trie in self.tries:
+            for n_gram, freq in trie.n_gram_frequencies.items():
+                decoded_n_gram = ''.join([self.storage.get_letter_by_id(letter_id)
+                                          for letter_id in n_gram])
+                freq_dict[decoded_n_gram] = freq
+        language_profile['freq'] = freq_dict
+        language_profile['n_words'] = self.n_words
+        language_profile['name'] = self.language
+        with open(name, 'w', encoding='utf-8') as file_to_save:
+            json_string = json.dumps(language_profile)
+            file_to_save.write(json_string)
```

Comment:
There is the `.dump` method in the `json` package that combines these two lines.